draft-ietf-rats-architecture.md

```diff
@@ -718,6 +718,13 @@ could establish trust in a Verifier as discussed above.
 
 ## Attester
 
+The Attester proves it is trustworthy to the Verifier by using key material provisioned into it by the Endorser.
```

Comment:
"provisioned into it by the Endorser" is not true in general.   For example, it's not true in the case where the Attester is an IoT device that generates its own key pair at the factory, and the public key is then extracted and used to generate a certificate issued by an Endorser (the manufacturer or whoever else), and the certificate itself isn't provisioned on the device but can be looked up on demand by the Verifier.  In this case, there is no "key material" provisioned into the Attester by any entity other than the Attester itself.

Line 723 and 754 are incorrect for the same reason.

Reply:
We probably should come up with better phrase -- "establish key material" or "set up key material" or "personalization"? It seems awkward to explain all the options each time it needs to be mentioned.

Even if the key material is generated on the device, you still have to do real work to 1) provide integrity when the public key is coming off the device and 2) make sure you are only taking keys off of real devices (no imposter, forgery or midnight runs). 

I have seen the term "provision" used to mean the whole process at a high level.

I know you keep bringing this up. I very much agree that what you suggest is the best way and should be allowed for and encouraged in the architecture doc, but I also know that are many situations where it can't be done. For example, it can't be done with symmetric keys. It can be done when you are putting keys into fuses on a chip that has no HW key generation feature. It can't be done when keys are supplied by someone other than the manufacturer.

So it seems we need some term or wording to describe this that covers all the scenarios.

Reply:
A common problem.

Maybe: "Attesters prove their trustworthiness to Verifiers via their key material that is established in their Attesting Environments under the control and authority of a trusted Endorsers." (might have an effect on the following sentences)